src/compute_accuracy.rs

```diff
@@ -243,8 +244,19 @@ fn read_analogies(reader: impl BufRead) -> Vec<Instance> {
 }
 
 fn process_analogies(embeddings: &Embeddings<VocabWrap, StorageViewWrap>, instances: &[Instance]) {
+    let pb = ProgressBar::new(instances.len() as u64);
+    pb.set_style(
+        ProgressStyle::default_bar().template("{bar:30} {percent}% {msg} ETA: {eta_precise}"),
+    );
     let eval = Eval::new(&embeddings);
     instances
         .par_iter()
-        .for_each(|instance| eval.eval_analogy(instance));
+        .enumerate()
+        .for_each(|(i, instance)| {
+            if i % 50 == 0 {
```

Comment:
Looking at this again, I want to verify that `i` is actually incremented by `1` per thread instead of being whatever the parallel iter just proceses. Otherwise we get updates whenever `i` happens to be a multiple of 50...

Reply:
From what I can tell, this does what we want it to do: enumerate each thread's iterator and updates the progress bar every 50 examples per thread. Printing 4 thread's `i` when the if statement is true results in:
~~~
0
4650
9300
13950
50
4700
9350
14000
100
4750
9400
14050
150
~~~
So every thread moves as expected.

Reply:
Maybe there is some documentation in rayon that describes the guarantees?

Reply:
Didn't find guarantees in the docs but the implementation seems to be that each thread iterates over its own range while processing examples:

https://docs.rs/rayon/1.2.0/src/rayon/iter/enumerate.rs.html#92-111

Reply:
Makes sense! Of course, it could happen that some thread gets index 50 immediately. But in the grand scheme of things, I think this should work reliably.

Reply:
I ran it a couple of times, the progressbar was well behaved. Although I haven't used windows for anything work-related in quite some time so I couldn't tell about that ;)